index.js

```diff
@@ -20,26 +20,35 @@ module.exports = function(source) {
     this.cacheable();
   }
   var done = this.async();
-  var options = loaderUtils.parseQuery(this.query);
+  var options = cloneDeep(loaderUtils.parseQuery(this.query) || {});
+
   options.dest = options.dest || '';
   options.filename = options.filename || this.resourcePath;
   options.Evaluator = CachedPathEvaluator;
-
   // Attach `importCache` to `options` so that the `Evaluator` can access it.
   var importCache = options.importCache = new ImportCache(this, options);
 
```

Comment:
Webpack 4 doesn’t provide `this.options`, see: https://github.com/shama/stylus-loader/commit/fc81e32e684dc95832d5ef0f24703265d1e132f6